packages/actor-query-operation-extend/lib/ActorQueryOperationExtend.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  ActorQueryOperation, ActorQueryOperationTypedMediated, Bindings,
+  IActorQueryOperationOutputBindings,
+  IActorQueryOperationTypedMediatedArgs,
+} from "@comunica/bus-query-operation";
+import { BindingsStream } from "@comunica/bus-query-operation";
+import { IActorTest } from "@comunica/core";
+import { termToString } from "rdf-string";
+import { Algebra } from "sparqlalgebrajs";
+import { AsyncEvaluator, ExpressionError } from "sparqlee";
+
+/**
+ * A comunica Extend Query Operation Actor.
+ */
+export class ActorQueryOperationExtend extends ActorQueryOperationTypedMediated<Algebra.Extend> {
+
+  constructor(args: IActorQueryOperationTypedMediatedArgs) {
+    super(args, 'extend');
+  }
+
+  public async testOperation(pattern: Algebra.Extend, context?: { [id: string]: any }): Promise<IActorTest> {
+    // will throw error for unsupported operators
+    const _ = new AsyncEvaluator(pattern.expression);
+    return true;
+  }
+
+  public async runOperation(pattern: Algebra.Extend, context?: { [id: string]: any })
+    : Promise<IActorQueryOperationOutputBindings> {
+
+    const { expression, input, variable } = pattern;
+
+    // Resolve the input
+    const output: IActorQueryOperationOutputBindings = ActorQueryOperation.getSafeBindings(
+      await this.mediatorQueryOperation.mediate({ operation: pattern.input, context }));
+    const bindingKey = termToString(variable);
+    const evaluator = new AsyncEvaluator(expression);
+    const throwIfHardError = (err: any) => {
+      if (!(err instanceof ExpressionError)) {
+        bindingsStream.emit('error', err);
+        throw err;
+      }
+    };
+
+    const transform = (item: Bindings, next: any) => {
+      evaluator.evaluate(item)
```

Comment:
I would use `await` and `try/catch` syntax for this block, as this will make it easier to read.